test/e2e/cluster_extension_install_test.go

```diff
@@ -644,7 +718,13 @@ func TestClusterExtensionInstallReResolvesWhenNewCatalog(t *testing.T) {
 		assert.Equal(ct, metav1.ConditionTrue, cond.Status)
 		assert.Equal(ct, ocv1alpha1.ReasonSuccess, cond.Reason)
 		assert.Contains(ct, cond.Message, "resolved to")
-		assert.Equal(ct, &ocv1alpha1.BundleMetadata{Name: "prometheus-operator.2.0.0", Version: "2.0.0"}, clusterExtension.Status.ResolvedBundle)
+		assert.Equal(ct,
+			&ocv1alpha1.ResolutionStatus{Bundle: &ocv1alpha1.BundleMetadata{
+				Name:    "prometheus-operator.1.2.0",
+				Version: "1.2.0",
```

Comment:
This is not a change I expect to see in a PR that is rearranging types. What's the background here?

Reply:
In order to compare the `BundleMetadata` to the new `Resolution.Bundle` spec, I had to make these changes.

Reply:
To clarify, since these are in an `Eventually` block, the e2e tests were panicking due to accessing the nil `clusterExtension.Status.Resolution` object to get the `Bundle` field. Doing this compares the entire `clusterExtension.Status.Resolution` object and avoids a potential panic. Alternatively an explicit `nil` check could be done before attempting to access the `Bundle` field.

Reply:
I should have been more specific in this comment (and it's no big deal since @tmshort fixed it in #1222), but what I was trying to ask is why this changed from 2.0.0 to 1.2.0.

I'm guessing that the `Eventually` aspect of this test is what tripped us up here. It is likely the case that the value is 1.2.0 when this check executes the first time, which explains why this would pass. However the purpose of this test is to see if a new catalog showing up under the same tag will trigger a re-reconcile and resolve a new version.

By setting this to 1.2.0, we were not actually testing what this test intended. Again, no big deal and the actual code was still behaving as expected, which is how Todd caught it.

However, I wonder: can we design our tests to be more robust against this kind of thing? Use of `Eventually` is likely the primary source of this problem because it implies both of:
- the current state of the field is not what we expect (otherwise, we would just directly check it)
- the value of the field we are asserting against will change over time.

And _that_ means that the test can pass using any assertion that passes on any initial or transitory state of that field.